pkg/endpoint/endpoint.go

```diff
@@ -420,9 +425,12 @@ func NewEndpointWithState(ID uint16, state string) *Endpoint {
 		state:         state,
 		hasBPFProgram: make(chan struct{}, 0),
 		controllers:   controller.NewManager(),
+		eventQueue:    newEventQueue(),
 	}
 	ep.SetDefaultOpts(option.Config.Opts)
 	ep.UpdateLogger(nil)
+	go ep.initializeEventQueue()
```

Comment:
initializing of the event queue should be done async inside the `sync.Once` inside the function